### PR TITLE
Incomplete coverage of test scenarios and optimize bad code

### PR DESCRIPTION
**Pull request by lunhuijie**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Increase test case scenarios and optimize code readability
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
NONE
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/util/pkiutil/pki_helpers.go [hunk 1]
```diff
@@ -247,13 +247,8 @@ func CertOrKeyExist(pkiPath, name string) bool {
...
 	_, certErr := os.Stat(certificatePath)
 	_, keyErr := os.Stat(privateKeyPath)
-	if os.IsNotExist(certErr) && os.IsNotExist(keyErr) {
-		// The cert and the key do not exist
-		return false
-	}
 
-	// Both files exist or one of them
-	return true
+	return !(os.IsNotExist(certErr) && os.IsNotExist(keyErr))
```
Comment (neolit123) [inline cmd/kubeadm/app/util/pkiutil/pki_helpers.go:251]: i think the original is better and provides slightly more clarity with the comments.
are you getting this as an error from a linter?

File: cmd/kubeadm/app/util/pkiutil/pki_helpers_test.go [hunk 2]
```diff
@@ -262,12 +262,16 @@ func TestCertOrKeyExist(t *testing.T) {
...
 			"failed WriteCertAndKey with an error: %v",
-			actual,
+			err,
+		)
+	}
+	if err = WriteCert(tmpdir, "foo-1", rootCACert); err != nil {
+		t.Errorf(
+			"failed WriteCert with an error: %v",
+			err,
 		)
```
Comment (neolit123) [inline cmd/kubeadm/app/util/pkiutil/pki_helpers_test.go:275]: i think these changes in pki_helpers_test are fine.

File: cmd/kubeadm/app/util/pkiutil/pki_helpers.go (hunk 1 above)
Comment (lunhuijie) [reply cmd/kubeadm/app/util/pkiutil/pki_helpers.go:251]: The next function of this function is written like this; I think this is more concise, and the logic itself is not complicated

Comment (lunhuijie) [reply cmd/kubeadm/app/util/pkiutil/pki_helpers.go:251]: so i change it by the way

Comment (neolit123) [reply cmd/kubeadm/app/util/pkiutil/pki_helpers.go:251]: > The next function of this function is written like this; I think this is more concise, and the logic itself is not complicated

ok, that is true.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
thanks

**Pull request merged**